spec.html

```diff
@@ -4906,6 +4980,89 @@ <h1>CopyDataProperties ( _target_, _source_, _excludedItems_ )</h1>
       </emu-note>
     </emu-clause>
 
+    <emu-clause id="sec-newprivatename" aoid="NewPrivateName">
+      <h1>NewPrivateName ( _description_ )</h1>
+      <emu-alg>
+        1. Return a new unique Private Name whose [[Description]] value is _description_.
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-privatefieldfind" aoid="PrivateFieldFind">
+      <h1>PrivateFieldFind ( _P_, _O_ )</h1>
+      <emu-alg>
+        1. Assert: _P_ is a Private Name.
+        1. Assert: _O_ is an object with a [[PrivateFieldValues]] internal slot.
+        1. For each element _entry_ in _O_.[[PrivateFieldValues]],
+          1. If _entry_.[[PrivateName]] is _P_, return _entry_.
+        1. Return ~empty~.
```

Comment:
since every time `PrivateFieldFind` is called, a TypeError is thrown when `~empty~` is returned, could we just make this operation throw the TypeError directly?

Reply:
`PrivateFieldAdd` throws when `PrivateFieldFind` does **not** return `~empty~`, so no.